skins/oasis/modules/OasisController.class.php

```diff
@@ -292,28 +371,48 @@ private function loadJs() {
 			$assetGroups[] = 'oasis_anon_js';
 		}
 
+
+		$jsLoader = '';
+
 		wfRunHooks('OasisSkinAssetGroups', array(&$assetGroups));
 
 		// add groups queued via OasisController::addSkinAssetGroup
 		$assetGroups = array_merge($assetGroups, self::$skinAssetGroups);
 
-		// SUS-905: Load all AssetsManager JS groups in a single HTTP request
-		/** @var SkinOasis $skin */
-		$skin = $this->getContext()->getSkin();
-		$jsLoader = $skin->getScriptsWithCombinedGroups( $assetGroups );
+		$assets = $this->assetsManager->getURL( $assetGroups ) ;
+
+		// jQueryless version - appears only to be used by the ad-experiment at the moment.
+		// disabled - not needed atm (and skipped in wsl-version anyway)
+		// $assets[] = $this->assetsManager->getURL( $isLoggedIn ? 'oasis_nojquery_shared_js_user' : 'oasis_nojquery_shared_js_anon' );
 
-		// These Wikia-specific scripts have already been output in the <head>
-		$this->wg->Out->topScripts = '';
-		$headScripts = $this->wg->Out->getHeadScripts();
+		// add $jsReferences
+		$assets = array_merge($assets, $jsReferences);
+
+		// generate direct script tags
+		foreach ($assets as $url) {
+			$url = htmlspecialchars( $url );
+			$jsLoader .= "<script src=\"{$url}\"></script>\n";
+		}
+
+		$tpl = $this->app->getSkinTemplateObj();
+
+		// $tpl->set( 'headscripts', $out->getHeadScripts() . $out->getHeadItems() );
+		// FIXME: we need to remove head items - i.e. <meta> tags
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this "FIXME" comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1134)